### DISCUSSION TITLE
How to create instrumentationf for maxQueries can be executed in single request?

### DISCUSSION THREAD

**Discussion by vrubal** [graphql-java/graphql-java#2937] [Ideas]
I'm using spring-graphql library. To add the maxQueryComplexity and maxQueryDepth instrumentation are available, which can be configured as follows. Is there any way to create a custom instrumentation to for maxQueries. 

@Bean
@ConditionalOnMissingBean
@ConditionalOnProperty(prefix = "spring.graphql.instrumentation", name = "max-query-complexity")
public MaxQueryComplexityInstrumentation maxQueryComplexityInstrumentation(@Value("${spring.graphql.instrumentation.max-query-complexity}") int maxComplexity) {
    return new MaxQueryComplexityInstrumentation(maxComplexity);
}

@Bean
@ConditionalOnMissingBean
@ConditionalOnProperty(prefix = "spring.graphql.instrumentation", name = "max-query-depth")
public MaxQueryDepthInstrumentation maxQueryDepthInstrumentation(@Value("${spring.graphql.instrumentation.max-query-depth}") int maxDepth) {
    return new MaxQueryDepthInstrumentation(maxDepth);
}

**Comment by vrubal** [comment]
Found a solution which seems to be working.

```
public class MaxQueryCountInstrumentation extends SimpleInstrumentation {

   private int maxQueryCount;
   public MaxQueryCountInstrumentation(int maxqueryCount) {
        this.maxQueryCount=maxqueryCount;
    }


    @Override
    public InstrumentationContext<ExecutionResult> beginExecuteOperation(InstrumentationExecuteOperationParameters parameters) {
        int size = ((OperationDefinition) parameters.getExecutionContext().getDocument().getDefinitions().get(0)).getSelectionSet().getSelections().size();
        if (maxQueryCount < size)
            throw new AbortExecutionException("Maximum query count exceeded " + size + " > " + maxQueryCount);
        return SimpleInstrumentationContext.noOp();
    }

}
```

**Reply by bbakerman** [reply]
self answered
